tests/installation/partitioning/warning/uefi_small.pm

```diff
@@ -0,0 +1,47 @@
+# SUSE's openQA tests
+#
+# Copyright © 2020 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Verify Warning Dialog when /boot/efi partition has too small
+# size.
+# Maintainer: QA SLE YaST team <qa-sle-yast@suse.de>
+
+use base 'y2_installbase';
+use strict;
+use warnings;
+use testapi;
+use scheduler 'get_test_suite_data';
+use Test::Assert ':all';
+
+my $partitioner;
+
+sub run {
+    my $test_data = get_test_suite_data();
+    my $disk      = $test_data->{disks}[0];
+    $partitioner = $testapi::distri->get_expert_partitioner();
+
+    $partitioner->run_expert_partitioner();
+    foreach my $partition (@{$disk->{partitions}->{uefi_small}}) {
+        $partitioner->add_partition_on_gpt_disk({
+                disk      => $disk->{name},
+                partition => $partition
+        });
+    }
+    $partitioner->accept_changes();
+
+    assert_matches(qr/$test_data->{warnings}->{uefi_small}/, $partitioner->get_warning_rich_text(),
+        "Warning Dialog for missing boot partition did not appear, while it is expected.");
+}
+
+sub post_run_hook {
```

Comment:
In case we don't want to make these modules fatal, what's about extracting this post run hook? As we have exactly same in `no_root` now.

Reply:
This post_run_hook I use as @After in JUnit and similar testing frameworks. Just to emphasize that test actually ended on `assert`. IMHO, introducing new namespace and make `$partitioner` variable public is also not a good solution. Inheritance in this case will be used in kind of artificial way. 

What I can do, but that is also not an ideal solution and I, personally, not a big fan of such kind of solutions, is to add method `decline_warning_and_cancel_changes` in ExpertPartitionerController. And then reuse it in `post_run_hook`. But methods, that has two actions in a name are also not a good pattern.

I would leave it as is for now, and let me finish remaining test modules for warnings. Then it will be more clear what could be reused and how can they be improved.

Reply:
Just as a sidenote, it's possible to use `$self` to pass the context and you already name `$partitioner` to be visible in the method. Anyway, let's keep it the way it is.